locations/storefinders/nomnom.py

```diff
@@ -0,0 +1,89 @@
+import re
+from datetime import date, timedelta
+from typing import Iterable
+from urllib import parse as urlparse
+
+from scrapy import Spider
+from scrapy.http import Request, Response, TextResponse
+
+from locations.categories import Extras, apply_yes_no
+from locations.dict_parser import DictParser
+from locations.hours import DAYS, DAYS_3_LETTERS, OpeningHours
+from locations.items import Feature
+
+
+def slugify(s: str) -> str:
+    return re.sub(r"\s+", "-", s).casefold()
+
+
+class NomNomSpider(Spider):
+    """NomNom is an "accelerator technology" by Bounteous for restaurants and
+    retailers:
+    https://www.bounteous.com/industries/restaurant-convenience/nomnom/
+    To use, specify "domain" as the base domain of the website."""
+
+    domain: str | None = None
+    use_calendar: bool = True
+
+    def _append_calendar_param(self, url):
+        if self.use_calendar:
+            parsed = urlparse.urlparse(url)
+            params = urlparse.parse_qs(parsed.query)
+            today = date.today()
+            params["nomnom"] = "calendars"
+            params["nomnom_calendars_from"] = [today.strftime("%Y%m%d")]
+            params["nomnom_calendars_to"] = [(today + timedelta(days=6)).strftime("%Y%m%d")]
+            parsed = parsed._replace(query=urlparse.urlencode(params, doseq=True))
+            return urlparse.urlunparse(parsed)
+        else:
+            return url
+
+    def start_requests(self) -> Iterable[Request]:
+        if self.start_urls:
+            for url in self.start_urls:
+                yield Request(self._append_calendar_param(url), dont_filter=True)
+        elif self.domain:
+            yield Request(self._append_calendar_param(f"https://nomnom-prod-api.{self.domain}/restaurants/"))
+
+    @staticmethod
+    def parse_opening_hours(calendar: dict) -> OpeningHours:
+        oh = OpeningHours()
+        for row in calendar["ranges"]:
+            if row["weekday"] not in DAYS_3_LETTERS:
+                continue
+            day = DAYS[DAYS_3_LETTERS.index(row["weekday"])]
+            oh.add_range(day, row["start"], row["end"], time_format="%Y%m%d %H:%M")
+        return oh
+
+    CALENDAR_KEYS = {
+        "drivethru": "opening_hours:drive_through",
+        "dispatch": "delivery",
```

Comment:
I'm not fond of this, [`delivery`](https://taginfo.openstreetmap.org/keys/delivery#values) is 99% classic OSM boolean. There is actually a conditional with more usage than traditional OSM opening hours.

[`opening_hours:delivery`](https://taginfo.openstreetmap.org/keys/opening_hours%3Adelivery#overview) has basically no usage, but I'd prefer it.

Reply:
Ok, I didn't look at actual usage, just documentation: https://wiki.openstreetmap.org/wiki/Key:delivery

Reply:
I'm really surprised by that.. idk